src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -486,6 +507,46 @@ private static Win32Program ExeProgram(string path)
             }
         }
 
+        // Function to get the application type, given the path to the application
+        public static ApplicationType GetAppTypeFromPath(string path)
```

Comment:
Can you add unit tests for this?  You may need to abstract the Directory and File methods

Reply:
I added an IDirectoryWrapper interface to Wox.Infrastructure (similar to IFileWrapper), and added tests for GetAppTypeFromPath.
I found that Directory.Exists was sufficient for checking of the path refers to a file or directory so I removed the code which checked FileAttributes.

For the mocked implementation for Directory.Exists, the logic I added was to check if the file path has an extension or not. This isn't actually true in Windows since you can have files without extensions, but I think for the purpose of a mock that is sufficient.